src/hooks/useStore.js

```diff
@@ -1,29 +1,132 @@
 import create from 'zustand';
 import * as Tone from 'tone';
+import { nanoid } from 'nanoid';
+import { defaultPadSettings } from '../data';
+import { defaultSequencerSettings } from '../data';
 
 const useStore = create(set => ({
   tone: null,
   dest: null,
   recorder: null,
+  loopPlayer: null,
+  currentDrumLoop: 'DrumLoop90BPM',
+  loopPlayerVolume: 5,
   recordings: [],
+  allPads: defaultPadSettings,
+  drumPadPlayersVolume: 5,
+  allPadSequences: defaultSequencerSettings,
+  selectedSequencerPad: 0,
+  currentTimeStamp: 0,
+  currentBpm: 100,
+  selectedPadSequence: defaultSequencerSettings[0].settings,
   isInstructionPopUpVisible: true,
   isInstructionOneVisible: false,
   isInstructionTwoVisible: false,
   isInstructionThreeVisible: false,
   isInstructionFourVisible: false,
   isInstructionFiveVisible: false,
-  addRecording: recording => {
-    set(state => ({ recordings: [recording, ...state.recordings] }));
-  },
+
   deleteRecording: recording => {},
-  
+
+  handleUserInteraction: () => {
+    const handleUserInteraction = () => {
+      const tone = useStore.getState().tone;
+      if (!tone) {
+        useStore.getState().initTone();
+        useStore.getState().initLoopPlayer();
+        useStore.getState().initDrumPadPlayers();
+      }
+    };
+    window.addEventListener('click', handleUserInteraction);
+    return () => {
+      window.removeEventListener('click', handleUserInteraction);
+    };
+  },
   initTone: () => {
     const tone = Tone.context;
+    tone._latencyHint = 'balanced';
     const dest = tone.createMediaStreamDestination();
     const recorder = new MediaRecorder(dest.stream);
+
     set({ tone, dest, recorder });
   },
-  
+
+  ///////     init DrumPadPlayers, set PadSettings and Volume    /////////
+  initDrumPadPlayers: () => {
+    const allPads = useStore.getState().allPads;
+    const drumPadPlayersVolume = useStore.getState().drumPadPlayersVolume;
+    const drumPadPlayers = new Tone.Players(
+      {
+        Player0: allPads[0].sample,
```

Comment:
Maybe you vould do something like that.`allPads.map((pad, index)=>'Player${index}:pad.sample)`

Reply:
Good idea. I will try that